tests/iot_config.h

```diff
@@ -124,17 +124,46 @@
     #define IOT_DEMO_PRIVATE_KEY           IOT_TEST_PRIVATE_KEY
 #endif
 #ifdef IOT_TEST_USER_NAME
-    #define IOT_DEMO_USER_NAME      IOT_TEST_USER_NAME
+    #define IOT_DEMO_USER_NAME             IOT_TEST_USER_NAME
 #endif
 #ifdef IOT_TEST_PASSWORD
-    #define IOT_DEMO_PASSWORD       IOT_TEST_PASSWORD
+    #define IOT_DEMO_PASSWORD              IOT_TEST_PASSWORD
 #endif
 #if defined( IOT_TEST_MQTT_CLIENT_IDENTIFIER )
     #define IOT_DEMO_IDENTIFIER            IOT_TEST_MQTT_CLIENT_IDENTIFIER
 #elif defined( AWS_IOT_TEST_SHADOW_THING_NAME )
     #define IOT_DEMO_IDENTIFIER            AWS_IOT_TEST_SHADOW_THING_NAME
 #endif
 
+/* Set defines required for Provisioning demo when building demo target with IOT_BUILD_TESTS set to 1. */
+#ifndef AWS_IOT_DEMO_PROVISIONING_TEMPLATE_NAME
+    #define AWS_IOT_DEMO_PROVISIONING_TEMPLATE_NAME                           ""
+#endif
```

Comment:
You're already defining `AWS_IOT_TEST_PROVISIONING_TEMPLATE_NAME` in this file. These lines should be:

```
#ifdef AWS_IOT_TEST_PROVISIONING_TEMPLATE_NAME
    #define AWS_IOT_DEMO_PROVISIONING_TEMPLATE_NAME    AWS_IOT_TEST_PROVISIONING_TEMPLATE_NAME
#endif